scalac-scoverage-plugin/src/main/scala/scoverage/report/ReportWriter.scala

```diff
@@ -0,0 +1,118 @@
+                   coverageOutputCobertura: Boolean,
+                   coverageOutputXML: Boolean,
+                   coverageOutputHTML: Boolean,
+                   coverageDebug: Boolean): Unit = {
+
+    if (coverageOutputCobertura) {
+      new CoberturaXmlWriter(baseDirectory, dir(outputDirectory, "cobertura")).write(coverage)
+    }
+
+    if (coverageOutputXML) {
+      val xmlDir = dir(outputDirectory, "xml")
+      new ScoverageXmlWriter(baseDirectory, xmlDir, false).write(coverage)
+      if (coverageDebug) {
+        new ScoverageXmlWriter(baseDirectory, xmlDir, true).write(coverage)
+      }
+    }
+
+    if (coverageOutputHTML) {
+      new ScoverageHtmlWriter(baseDirectory, dir(outputDirectory, "html")).write(coverage)
+    }
+
+  }
+}
+
+object Reports extends App {
+
+  private val defaultsToTrue = () => Some(true)
+  private implicit val stringToFile = singleArgConverter(new File(_))
+
+  val BASE_DIR: String = "base-dir"
+  val OUTPUT_DIR: String = "output-dir"
+  val AGGREGATE_SUB_PROJECTS: String = "aggregate"
+  val CLEAN_AGGREGATED_FILES: String = "clean"
+  val SINGLE_DATA_DIR: String = "data-dir"
+  val OUTPUT_COBERTURA_XML: String = "cobertura"
+  val OUTPUT_SCOVERAGE_DEBUG_XML: String = "debug"
+  val OUTPUT_SCOVERAGE_XML: String = "xml"
+  val OUTPUT_SCOVERAGE_HTML: String = "html"
+
+  private val opts = Scallop(args)
+    .opt[File](BASE_DIR, required = true)
+    .opt[File](OUTPUT_DIR, required = true)
+    .opt[Boolean](AGGREGATE_SUB_PROJECTS, required = false, default = () => None)
+    .opt[File](SINGLE_DATA_DIR, required = false, validate = f => Serializer.coverageFile(f).exists())
+    .opt[Boolean](CLEAN_AGGREGATED_FILES, default = defaultsToTrue)
+    .opt[Boolean](OUTPUT_SCOVERAGE_DEBUG_XML, default = defaultsToTrue)
+    .opt[Boolean](OUTPUT_COBERTURA_XML, default = defaultsToTrue)
+    .opt[Boolean](OUTPUT_SCOVERAGE_XML, default = defaultsToTrue)
+    .opt[Boolean](OUTPUT_SCOVERAGE_HTML, default = defaultsToTrue)
+    .verify
+
+  private val baseDir = opts[File](BASE_DIR)
+
+  private val coverage: Option[Coverage] = {
+    // opts.get[_] appears not to work as expected here
+    (opts.isSupplied(SINGLE_DATA_DIR), opts.isSupplied(AGGREGATE_SUB_PROJECTS)) match {
+      case (true, false) =>
+        val dataDir = opts[File](SINGLE_DATA_DIR)
+        val coverageFile = Serializer.coverageFile(dataDir)
+        if (coverageFile.exists()) {
+          val coverage = Serializer.deserialize(coverageFile)
+          val measurementFiles = IOUtils.findMeasurementFiles(dataDir)
+          val measurements = IOUtils.invoked(measurementFiles)
+          coverage.apply(measurements)
+          Some(coverage)
+        } else {
+          None
+        }
+      case (false, true) =>
+        CoverageAggregator.aggregate(baseDir, opts[Boolean](CLEAN_AGGREGATED_FILES))
+      case (true, true)=>
+        Console.err.println(s"Cannot specify $SINGLE_DATA_DIR and $AGGREGATE_SUB_PROJECTS")
```

Comment:
I like to have messages prefixed with plugin name, like `[scoverage] Cannot specify (...)`, what do you think about that?